experimental/telemetry/tracker.go

```diff
@@ -66,20 +99,42 @@ func NewTracker(
 		serverVersion:      serverVersion,
 		pluginID:           pluginID,
 		pluginVersion:      pluginVersion,
-		enabled:            enableDiagnostics,
+		enabledLogging:     config.EnabledLogging,
+		enabledTracking:    config.EnabledTracking,
+		logger:             l,
 	}
 }
 
+func (t *tracker) ReloadConfig(config TrackerConfig) {
```

Comment:
Nit pick:

The plugin code currently calls this method like this:
```golang
	if p.tracker != nil {
		p.tracker.ReloadConfig(telemetry.NewTrackerConfig(p.API))
	}
```

If the code should support the case that `tracker` is `nil`, it could do so by returning early:
```suggestion
func (t *tracker) ReloadConfig(config TrackerConfig) {
	if t == nil {
		return
	}
```

Reply:
I've tried it but it ends up in a panic.

I think that since we work with the Tracker interface, when the realtracker still doesn't exist there's no type to cast to, and therefore it's pure nil instead of typed-nil.


https://play.golang.com/p/FrES0m9IV6X

Reply:
Good catch. I missed that `NewTracker` returns an interface instead of a concrete type. :+1: 

A bit off-topic: Per the proverb ("Accept interfaces, return structs")[https://bryanftan.medium.com/accept-interfaces-return-structs-in-go-d4cab29a301b] I'm wondering if we should kill the `Tracker` interface entirely. That would also solve the `nil` issue.

Reply:
good point and agree with the overall approach about structs/intertfaces you mentioned.

however, to continue changing this means to make public tracker struct + choose a new name to avoid conflict with interface (or skipping revive linting rule). I'd keep that change out of this PR unless you think it's crucial.

Reply:
I'm fine with keeping that conversation out of this PR. :ok_hand: